File: deque/deque_.py
class Deque:
    def __init__(self):
        self.dequeue = []

    def addFront(self, item):
        self.dequeue.insert(0, item)

    def addTail(self, item):
        self.dequeue.append(item)

    def removeFront(self):
        if not self.dequeue:
            return None

        return self.dequeue.pop(0)

    def removeTail(self):
        if not self.dequeue:
            return None
        return self.dequeue.pop()

    def size(self):
        return len(self.dequeue)

File: asd_go/deque/deque.go
package main

import (
	"container/list"
	"errors"
	"os"
)

type Deque[T any] struct {
	lst *list.List
}

func (d *Deque[T]) Size() int {
	d.init()

	return d.lst.Len()
}

func (d *Deque[T]) AddFront(itm T) {
	d.init()

	d.lst.PushFront(itm)
}

func (d *Deque[T]) AddTail(itm T) {
	d.init()

	d.lst.PushBack(itm)
}

func (d *Deque[T]) RemoveFront() (T, error) {
	var result T
	if d.Size() == 0 {
		return result, errors.New("empty queue")
	}

	untyped := d.lst.Remove(d.lst.Front())
	result, ok := untyped.(T)
	if !ok {
		return result, errors.New("unsupported type")
	}

	return result, nil
}

func (d *Deque[T]) RemoveTail() (T, error) {
	var result T
	if d.Size() == 0 {
		return result, errors.New("empty queue")
	}

	untyped := d.lst.Remove(d.lst.Back())
	result, ok := untyped.(T)
	if !ok {
		return result, errors.New("unsupported type")
	}

	return result, nil
}

func (d *Deque[T]) init() {
	if d.lst == nil {
		d.lst = list.New()
	}
}
